### ISSUE TITLE
[Decoder/Test] Different output file with valgrind

### ISSUE THREAD

**Issue by kparichay** [nnstreamer/nnstreamer#1367]
Running ssat test cases for tensor_decoder with valgrind fails the Golden Test 1-1 testcase.
The reason is the output file generated `testcase01_RGB_642x480.log` differs from the `testcase01_RGB_642x480.golden.raw` and `callCompareTest()` fails. Note that the `gstTest()` call for generating both the file does not fail and returns with success return code.

Can you please check it.
For running with valgrind, add `valgrind --track-origins=yes` as a prefix in `gstTest()`

Note that valgrind shows uninitialized memory error - it can give you some hint or you can ignore it as there are many false positives in valgrind and these error can be suppressed.

(Note by MJ)
Expected worksize < 1 mw (supposed to be a simple debugging)

**Comment by kparichay** [comment]
valgrind has been added to ssat, you can directly run with valgrind by passing `--enable-valgrind` option to ssat.

**Comment by kparichay** [comment]
It might be possible that the issue might be arising from tensor_converter as the error happens only in the case of test image RGB 642x480 (when `offset % 4 != 0` as per line 626 of tensor_converter.c).
@myungjoo can you please take a look.

**Comment by kparichay** [comment]
@jijoongmoon I have verified the code for tensor_converter and it looks good to me. 
Moreover, the test cases of tensor_converter cover the case with 642x480 RGB image and there is no error in it.

Can you please check the code in tensor_decoder.c

**Comment by myungjoo** [comment]
Isn't this resolved?

**Comment by kparichay** [comment]
Not yet resolved
@jijoongmoon Can you please take a look

**Comment by myungjoo** [comment]
@jijoongmoon may start looking at this after other more urgent issue is resolved (probably after 1st October)

**Comment by myungjoo** [comment]
He is getting more and more urgent tasks. This issue is available for anyone.

**Comment by gichan-jang** [comment]
@kparichay, I think this issue has been resolved with #3626. Please check and close the issue.

```
gichan@gichan:~/workspace/nnstreamer/tests/nnstreamer_decoder$ ssat --enable-valgrind
==================================================
    Test Group nnstreamer_decoder Starts.
[PASSED] 1_gen_golden_data:gst-launch of case 1_gen_golden_data
[PASSED] 1:gst-launch of case 1
[PASSED] 1:Golden Test 1 comparison
[PASSED] 1-1_gen_golden_data:gst-launch of case 1-1_gen_golden_data
[PASSED] 1-1:gst-launch of case 1-1
[PASSED] 1:Golden Test 1-1 comparison
[PASSED] 1-2_gen_golden_data:gst-launch of case 1-2_gen_golden_data
[PASSED] 1-2:gst-launch of case 1-2
[PASSED] 1:Golden Test 1-2 comparison
[PASSED] 1-3_gen_golden_data:gst-launch of case 1-3_gen_golden_data
[PASSED] 1-3:gst-launch of case 1-3
[PASSED] 1:Golden Test 1-3 comparison
[PASSED] 2:gst-launch of case 2
[PASSED] 2:Compare for case 2
[PASSED] 3:gst-launch of case 3
[PASSED] 3:Compare for case 3
[PASSED] 4F_n:gst-launch of case 4F_n
[PASSED] 5:gst-launch of case 5
[PASSED] 5-1:Compare for case 5-1
[PASSED] 5-2:Compare for case 5-2
[PASSED] 5-3:Compare for case 5-3
[PASSED] 6:gst-launch of case 6
[PASSED] 6-1:Compare for case 6-1
[PASSED] 6-2:Compare for case 6-2
[PASSED] 6-3:Compare for case 6-3
==================================================
[PASSED] Test Group nnstreamer_decoder Passed



==================================================

==================================================

[PASSED] nnstreamer_decoder (25 passed among 25 cases)
==================================================
[PASSED] All Test Groups (1) Passed!
         TC Passed: 25 / Failed: 0 / Ignored: 0 
```
